packages/flutter_tools/lib/src/device.dart

```diff
@@ -294,19 +312,123 @@ abstract class DeviceManager {
 
     return devices;
   }
+}
+
+/// A class for determining how to filter devices based on if they are supported.
+///
+/// If [mustBeSupportedByFlutter] is true, only devices supported by Flutter
+/// will be included.
+///
+/// If [mustBeSupportedForProject] is true, only devices supported by Flutter
+/// and supported for the provided [flutterProject] will be included.
+///
+/// If [mustBeSupportedForAll] is true, only devices supported by Flutter,
+/// supported for the provided [flutterProject], and supported for
+/// `--device all` will be included.
+class DeviceDiscoverySupportFilter {
```

Comment:
Yeah the initial reason was because I made `flutterProject` required in `DeviceDiscoverySupportFilter`, but `DeviceDiscoveryFilter` can be used without filtering by support and other filters in `DeviceDiscoveryFilter` did not need `flutterProject`. With my latest update, another reason is because it needs to be overridable for g3.

Reply:
Thanks, makes sense!